### ISSUE TITLE
`AttributeError: 'NoneType' object has no attribute '_spec_coroutines'` error when creating `asynctest.CoroutineMock()` in `pytest`

### ISSUE THREAD

**Issue by IlyaMichlin** [DataDog/dd-trace-py#4484]
### Summary of problem

Getting `AttributeError: 'NoneType' object has no attribute '_spec_coroutines'` error when creating `asynctest.CoroutineMock()` in `pytest` test.

The code provided below produces the error only when `ddtrace==1.6.0` is installed. If `ddtrace==1.5.3` is installed or no `ddtrace` is installed, the error does not accure.

After some debugging, I found that there is a difference in the output of `dir(None)` which ultimately causes the error.

When `ddtrace==1.5.3` or no `ddtrace` is installed:
`dir(None) = ['__bool__', '__class__', '__delattr__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__']`

When `ddtrace==1.6.0` is installed:
`dir(None) = ['ANY', 'AsyncMagicMixin', 'CoroutineMock', 'DEFAULT', 'FakeInheritanceMeta', 'GLOBAL', 'IsCoroutineArgMeta', 'LIMITED', 'MagicMock', 'Mock', 'MockMetaMixin', 'NonCallableMagicMock', 'NonCallableMock', 'PatchScope', 'PropertyMock', '_AsyncIterator', '_AwaitEvent', '_PatchedGenerator', '__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__spec__', '_async_magics', '_clear_dict', '_decorate_coroutine_callable', '_get_async_iter', '_get_child_mock', '_get_is_coroutine', '_is_coroutine', '_is_started', '_make_native_coroutine', '_mock_add_spec', '_patch', '_patch_dict', '_patch_multiple', '_patch_object', '_pydev_stop_at_break', '_raise', '_set_is_coroutine', '_spec_coroutines', '_update_new_callable', 'args', 'async_magic_coroutines', 'asyncio', 'attr', 'call', 'contextlib', 'create_autospec', 'enum', 'functools', 'inspect', 'kwargs', 'mock_open', 'patch', 'return_once', 'self', 'sentinel', 'spec', 'sys', 'types', 'unittest']`

### Which version of dd-trace-py are you using?
`ddtrace==1.6.0`

### Which version of pip are you using?
`pip==22.3.1`

### Which libraries and their versions are you using?
<details>
  <summary>`pip freeze not working`</summary>
  asynctest==0.13.0
  attrs==22.1.0
  bytecode==0.13.0
  cattrs==22.2.0
  ddsketch==2.0.4
  ddtrace==1.6.0
  envier==0.4.0
  exceptiongroup==1.0.1
  forbiddenfruit==0.1.4
  importlib-metadata==5.0.0
  importlib-resources==5.10.0
  iniconfig==1.1.1
  jsonschema==4.17.0
  packaging==21.3
  pkgutil_resolve_name==1.3.10
  pluggy==1.0.0
  protobuf==4.21.9
  pyparsing==3.0.9
  pyrsistent==0.19.2
  pytest==7.2.0
  six==1.16.0
  tenacity==8.1.0
  tomli==2.0.1
  typing_extensions==4.4.0
  xmltodict==0.13.0
  zipp==3.10.0
</details>

<details>
  <summary>`pip freeze working`</summary>
  
  asynctest==0.13.0
  attrs==22.1.0
  exceptiongroup==1.0.1
  importlib-metadata==5.0.0
  iniconfig==1.1.1
  packaging==21.3
  pluggy==1.0.0
  pyparsing==3.0.9
  pytest==7.2.0
  tomli==2.0.1
  typing_extensions==4.4.0
  zipp==3.10.0

</details>

### How can we reproduce your problem?
To reproduce this error you only need to install the 3 libraries and running the `test_t` test:
```
import asynctest
asynctest.CoroutineMock()


def test_t():
    pass
```

### What is the result that you get?
Traceback:
```
tests/test_t.py:None (tests/test_t.py)
tests/test_t.py:2: in <module>
    asynctest.CoroutineMock()
venv/lib/python3.7/site-packages/asynctest/mock.py:572: in __init__
    super().__init__(*args, **kwargs)
/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/unittest/mock.py:1001: in __init__
    _spec_state, _new_name, _new_parent, **kwargs
/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/unittest/mock.py:411: in __init__
    self._mock_add_spec(spec, spec_set, _spec_as_instance, _eat_self)
venv/lib/python3.7/site-packages/asynctest/mock.py:181: in _mock_add_spec
    if asyncio.iscoroutinefunction(getattr(spec, attr)):
E   AttributeError: 'NoneType' object has no attribute '_spec_coroutines'
```

**Comment by majorgreys** [comment]
Also, can you share details on whether and how ddtrace is being loaded in the execution of your tests?

**Comment by majorgreys** [comment]
I was able to reproduce this with the following files copied over to a clean Docker image for Python 3.7:

```
docker run -v $(PWD)/:/workdir -w /workdir -it --rm python:3.7 ./run.sh
```

https://gist.github.com/majorgreys/7cb81d880286fdf278a66bdb6032e5f7

**Comment by bparker98** [comment]
The real issue here is that `forbiddenfruit` has an import side-effect where it replaces the builtin `dir()` function with a new version that does not behave as expected.

After importing `forbiddenfruit` the `dir()` function returns the wrong values when doing `dir(None)`, it returns stuff from the local call frame.

This is an example script that shows this bad behavior.

```
from forbiddenfruit import curse

def example():
    foo = 0
    bar = 1
    for attr in dir(None):
        print(f"{attr} = {getattr(None, attr)}")

def lazy_example():
    from forbiddenfruit import curse
    example()


example()
lazy_example()
```

This will fail when calling `lazy_example()` because the behavior of `dir()` has changed.

I think this is not expected, and could have unintended consequences. I'm also not a big fan of changing immutable builtins to begin with.

**Comment by bparker98** [comment]
The code in question for `forbiddenfruit` can be found here. https://github.com/clarete/forbiddenfruit/blob/master/forbiddenfruit/__init__.py#L251

**Comment by juanjux** [comment]
Should be fixed in 1.6.1 that will release soon.
